spec/acceptance/auth_controller_spec.rb

```diff
@@ -265,6 +265,173 @@
 
         expect(response_body).to include("invalid_token")
       end
+
+      context "facebook ID token (OIDC)" do
+        let(:facebook_id_token) { "facebook_oidc_test_token" }
+
+        before do
+          # Set up default JWT mock to avoid conflicts with other JWT usage in the app
+          allow(JWT).to receive(:decode).and_call_original
+
+          # Mock the JWKS fetching
+          allow(Net::HTTP).to receive(:get_response).with(URI("https://limited.facebook.com/.well-known/openid-configuration/"))
+            .and_return(double("response", is_a?: Net::HTTPSuccess, body: '{"jwks_uri": "https://www.facebook.com/.well-known/oauth/jwks"}'))
+
+          allow(Net::HTTP).to receive(:get_response).with(URI("https://www.facebook.com/.well-known/oauth/jwks"))
+            .and_return(double("response", is_a?: Net::HTTPSuccess, body: '{"keys": [{"kty": "RSA", "kid": "test"}]}'))
+
+          # Now create a specific mock for our Facebook OIDC token
+          facebook_jwt_options = {
```

Comment:
[Nice-to-have] facebook_jwt_options hash duplicated across multiple test cases. Evidence: Lines 284-291, 345-352, 366-373, 387-394, 408-415. Impact: Code duplication increases maintenance. Suggestion: Extract to let block. [Category: File]